src/aind_registration_evaluation/util/file_conversion.py

```diff
@@ -0,0 +1,257 @@
+    if not (os.path.isfile('/scratch/confocal_s0_cropped_downsampled.tif')):
+        downsample_and_int16('/scratch/confocal_s0_cropped.tif','/scratch/confocal_s0_cropped_downsampled.tif', downsampling_factor)
+    if not (os.path.isfile('/scratch/cortical_s0_cropped_downsampled.tif')):
+        downsample_and_int16('/scratch/cortical_s0_cropped.tif','/scratch/cortical_s0_cropped_downsampled.tif', downsampling_factor)
+
+
+def tiff_to_n5(tiff_filename, n5_filename, downsampling_factors):
+    '''
+    converts tiff to n5 with the specified resolution
+
+    Parameters
+        ------------------------
+        tiff_filename: input file path (string)
+        n5_filename: out file path (string)
+        downsampling_factor: 2**downsampling factor to change resolution
+
+        Returns
+        ------------------------
+        none
+
+    '''
+    image_volume = tifffile.imread(tiff_filename)
+    shape = image_volume.shape
+    # Create an .n5 dataset
+    f = z5py.File(n5_filename)
+    # Create a group for the multiscale data
+    if "c0" not in f:
+        group = f.create_group("c0")
+    else:
+        group = f["c0"]
+
+    # Create datasets for each scale level
+    for i, factor in enumerate(downsampling_factors):
+        # Calculate downsampled shape
+        if "s"+str(i) in group:
+            del group["s"+str(i)] 
+        ds_data = image_volume[::factor, ::factor, ::factor]
+        downsampled_shape = ds_data.shape
+        ds = group.create_dataset('s'+str(i), shape=downsampled_shape,dtype=image_volume.dtype, chunks=(128, 128, 128), compression="gzip")
+        ds[:] = ds_data
+        # Optionally, store additional metadata
+        ds.attrs["downsamplingFactors"] = tuple([factor,factor,factor])  # Example: store resolution
+
+    # Close the .n5 file
+    f.close()
+
+
+def n5_to_tiff(n5_filename, output_path, scale_level):
+    '''
+    converts n5 to to tif with the specified resolution
+
+    Parameters
+        ------------------------
+        tiff_filename: out file path (string)
+        n5_filename: input file path (string)
+        downsampling_factor: pyramid level of resolution
+
+        Returns
+        ------------------------
+        none
+
+    '''
+    f = z5py.File(n5_filename)
+    image = np.array(f["c0"][scale_level])
+    tifffile.imwrite(output_path, image)
+
+def n5_to_tiff_warped( n5_filename, output_path):
+    '''
+    converts the warped n5 to tiff (comes without resolution scale) in full resolution
+
+    Parameters
+        ------------------------
+        tiff_filename: out file path (string)
+        n5_filename: input file path (string)
+
+        Returns
+        ------------------------
+        none
+
+    '''
+    f = z5py.File(n5_filename)
+    image = np.array(f["c0"])
+    tifffile.imwrite(output_path, image)
+
+def apply_mask(inFilepath, outFilepath, threshhold, newValue):
```

Comment:
the file is named file_conversion.py but some processing functions are included here. Please, move the function where it's convenient.